File: 백준/Silver/10866. 덱/덱.py
import sys
n = int(input())
deque = []

for i in range(n):
    user = sys.stdin.readline().split()

    if user[0] == 'push_front':
        deque.insert(0, int(user[1]))

    elif user[0] == 'push_back':
        deque.append(int(user[1]))

    elif user[0] == 'pop_front':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque.pop(0))

    elif user[0] == 'pop_back':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque.pop())

    elif user[0] == 'size':
        print(len(deque))

    elif user[0] == 'empty':
        if len(deque) == 0:
            print(1)
        else:
            print(0)

    elif user[0] == 'front':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[0])

    elif user[0] == 'back':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[-1])

File: 백준/Silver/10866. 덱/덱.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const N = input[0];
let deque = [];
let ans = '';

for (let i = 1; i <= N; i++) {
    if (input[i].split(' ').length === 1) {
        if (input[i] === 'size') {
            ans += deque.length + '\n';
        } else if (input[i] === 'empty') {
            if (deque.length === 0) {
                ans += '1\n';
            } else {
                ans += '0\n';
            }
        } else if (input[i] === 'front') {
            if (deque.length === 0) {
                ans += '-1\n';
            } else {
                ans += deque[0] + '\n';
            }
        } else if (input[i] === 'back') {
            if (deque.length === 0) {
                ans += '-1\n';
            } else {
                ans += deque[deque.length - 1] +'\n';
            }
        } else if (input[i] === 'pop_front') {
            if (deque.length === 0) {
                ans += '-1\n';
            } else {
                ans += deque.shift() + '\n';
            }
        } else if (input[i] === 'pop_back') {
            if (deque.length === 0) {
                ans += '-1\n';
            } else {
                ans += deque.pop() + '\n';
            }
        }
    } else {
        if (input[i].split(' ')[0] === 'push_front') {
            deque.unshift(input[i].split(' ')[1]);
        } else {
            deque.push(input[i].split(' ')[1]);
        }
    }
}
console.log(ans);